common/Geometry2d/Point.hpp

```diff
@@ -12,75 +12,65 @@ Simple class to represent a point in 2d space. Uses floating point coordinates
 */
 class Point {
 public:
-    /** the x coordinate */
-    float x;
-
-    /** the y coordinate */
-    float y;
+    const float& x() const { return _x; }
```

Comment:
Using `inline` for this purpose is more or less deprecated - any reasonable compiler will inline this automatically.